client/app/components/navbar/navbar-notification.service.js

```diff
@@ -0,0 +1,109 @@
+        this.UNIVERSE = UNIVERSE;
+
+        this.NOTIFICATION_REFRESH_TIME = 60000;
+    }
+
+    getMessages () {
+        return this.$translate.refresh().then(() => this.OvhApiNotificationAapi.query({
+            lang: this.$translate.preferredLanguage(),
+            target: this.constants.target,
+            universe: this.UNIVERSE
+        }).$promise);
+    }
+
+    getSubLinks () {
+        return this.getMessages().then((messages) => messages.map((message) => this.convertSubLink(message))).catch(() => [{
+            id: "0",
+            date: moment(),
+            time: this._formatTime(moment()),
+            level: "error",
+            subject: this.translator.tr("common_navbar_notification_error_subject"),
+            description: this.translator.tr("common_navbar_notification_error")
+        }]);
+    }
+
+    /* eslint-disable class-methods-use-this */
+    _formatTime (dateTime) {
+        return moment(dateTime).fromNow();
+    }
+    /* eslint-enable class-methods-use-this */
+
+    _toggleSublinkAction (toUpdate, linkClicked) {
+        if (toUpdate.isActive && !toUpdate.updating) {
+            toUpdate.updating = true;
+            this.OvhApiNotificationAapi.post({ completed: [toUpdate.id] }).$promise.then(() => {
+                toUpdate.isActive = false;
+                toUpdate.acknowledged = true;
+            }).finally(() => { toUpdate.updating = false; });
+        } else if (!toUpdate.isActive && !toUpdate.updating && !linkClicked) {
+            toUpdate.updating = true;
+            this.OvhApiNotificationAapi.post({ acknowledged: [toUpdate.id] }).$promise.then(() => {
+                toUpdate.isActive = true;
+                toUpdate.acknowledged = true;
+            }).finally(() => { toUpdate.updating = false; });
+        }
+    }
+
+    convertSubLink (notification) {
+        notification.time = this._formatTime(notification.date);
+        notification.url = notification.urlDetails.href;
+        notification.isActive = _.contains(["acknowledged", "delivered"], notification.status);
+        notification.acknowledged = _.contains(["acknowledged", "completed", "unknown"], notification.status);
+        notification.actionClicked = (toUpdate) => this._toggleSublinkAction(toUpdate);
+        notification.linkClicked = (toUpdate) => this._toggleSublinkAction(toUpdate, true);
+        return notification;
+    }
+
+    acknowledgeAll () {
+        if (this.navbarContent) {
+            const toAcknowledge = this.navbarContent.subLinks
+                .filter((subLink) => !subLink.acknowledged && subLink.isActive);
+            if (toAcknowledge.length) {
+                this.OvhApiNotificationAapi.post({ acknowledged: toAcknowledge.map((x) => x.id) }).$promise.then(() => {
+                    toAcknowledge.forEach((sublink) => {
+                        sublink.acknowledged = true;
+                    });
+                });
+            }
+        }
+    }
+
+    _setRefreshTime (sublinks) {
```

Comment:
Maybe you would refresh the elapsed time only when you open the notification center ? No need to compute time elapsed if nobody looks at it

Reply:
Also, shouldn't it be done by the component itself ? The implementation would be the same, disregard the manager you're in

Reply:
I don't want to add moment dependancy to ovh-ui-angular